Compilation fails

#Description

The compilation does not work throwing the following errors:


``` bash
cc -std=gnu99 -O3 -Wall   -c -o xvcpi.o xvcpi.c
cc -o xvcpi -lpigpio -lrt xvcpi.o
/usr/bin/ld: xvcpi.o: in function `stop_running':
xvcpi.c:(.text+0x0): undefined reference to `gpioTerminate'
/usr/bin/ld: xvcpi.o: in function `handle_data':
xvcpi.c:(.text+0x218): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x224): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x230): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x2d8): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x2e4): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.o:xvcpi.c:(.text+0x2f0): more undefined references to `gpioWrite' follow
/usr/bin/ld: xvcpi.o: in function `handle_data':
xvcpi.c:(.text+0x304): undefined reference to `gpioRead'
/usr/bin/ld: xvcpi.c:(.text+0x318): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x324): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x330): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x3e8): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x3f4): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.o:xvcpi.c:(.text+0x400): more undefined references to `gpioWrite' follow
/usr/bin/ld: xvcpi.o: in function `handle_data':
xvcpi.c:(.text+0x4d4): undefined reference to `gpioRead'
/usr/bin/ld: xvcpi.c:(.text+0x4f0): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x4fc): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text+0x508): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.o: in function `main':
xvcpi.c:(.text.startup+0xcc): undefined reference to `gpioInitialise'
/usr/bin/ld: xvcpi.c:(.text.startup+0xe8): undefined reference to `gpioSetSignalFunc'
/usr/bin/ld: xvcpi.c:(.text.startup+0xfc): undefined reference to `gpioSetMode'
/usr/bin/ld: xvcpi.c:(.text.startup+0x108): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text.startup+0x114): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text.startup+0x120): undefined reference to `gpioWrite'
/usr/bin/ld: xvcpi.c:(.text.startup+0x12c): undefined reference to `gpioSetMode'
/usr/bin/ld: xvcpi.c:(.text.startup+0x138): undefined reference to `gpioSetMode'
/usr/bin/ld: xvcpi.c:(.text.startup+0x144): undefined reference to `gpioSetMode'
/usr/bin/ld: xvcpi.c:(.text.startup+0x4ec): undefined reference to `gpioTerminate'
/usr/bin/ld: xvcpi.c:(.text.startup+0x504): undefined reference to `gpioTerminate'
/usr/bin/ld: xvcpi.c:(.text.startup+0x518): undefined reference to `gpioTerminate'
/usr/bin/ld: xvcpi.c:(.text.startup+0x52c): undefined reference to `gpioTerminate'
collect2: error: ld returned 1 exit status
make: *** [Makefile:7: xvcpi] Error 1
```

# Solution

It appears that there the library order of dependency is not correct.

Instead of:
``` Makefile
...
$(PROG): $(PROG).o
        $(CC) -o $(PROG) -lpigpio -lrt $<
...
```

Do:


``` Makefile
$(PROG): $(PROG).o
        $(CC) -o $(PROG)  $< -lpigpio -lrt
```

--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ CFLAGS=-std=gnu99 -O3 -Wall
 all: $(PROG)
 
 $(PROG): $(PROG).o
-	$(CC) -o $(PROG) -lpigpio -lrt $<
+	$(CC) -o $(PROG) $< -lpigpio -lrt
 
 clean:
 	rm -f $(PROG) *.o

